ui/tests/integration/components/auth-jwt-test.js

```diff
@@ -172,84 +176,77 @@ module('Integration | Component | auth jwt', function(hooks) {
     this.set('selectedAuthPath', 'foo');
     await component.role('test');
     component.login();
-
-    later(async () => {
-      run.cancelTimers();
-      await settled();
-      let call = this.window.open.getCall(0);
-      assert.deepEqual(
-        call.args,
-        [
-          'http://example.com',
-          'vaultOIDCWindow',
-          'width=500,height=600,resizable,scrollbars=yes,top=0,left=0',
-        ],
-        'called with expected args'
-      );
-    }, 50);
-    await settled();
+    await waitUntil(() => {
+      return this.openSpy.calledOnce;
+    });
+    run.cancelTimers();
+    let call = this.openSpy.getCall(0);
+    assert.deepEqual(
+      call.args,
+      ['http://example.com', 'vaultOIDCWindow', 'width=500,height=600,resizable,scrollbars=yes,top=0,left=0'],
+      'called with expected args'
+    );
   });
 
   test('oidc: it calls error handler when popup is closed', async function(assert) {
     await renderIt(this);
     this.set('selectedAuthPath', 'foo');
     await component.role('test');
     component.login();
-
-    later(async () => {
-      this.window.close();
-      await settled();
-      assert.equal(this.error, ERROR_WINDOW_CLOSED, 'calls onError with error string');
-    }, 50);
+    await waitUntil(() => {
+      return this.openSpy.calledOnce;
+    });
+    this.window.close();
     await settled();
+    assert.equal(this.error, ERROR_WINDOW_CLOSED, 'calls onError with error string');
   });
 
   test('oidc: storage event fires with wrong key', async function(assert) {
     await renderIt(this);
     this.set('selectedAuthPath', 'foo');
     await component.role('test');
     component.login();
-    later(async () => {
-      run.cancelTimers();
-      this.window.trigger('storage', { key: 'wrongThing' });
-      assert.equal(this.window.localStorage.removeItem.callCount, 0, 'never calls removeItem');
-    }, 50);
-    await settled();
+    await waitUntil(() => {
+      return this.openSpy.calledOnce;
+    });
+    this.window.trigger('storage', { key: 'wrongThing' });
+    assert.equal(this.window.localStorage.removeItem.callCount, 0, 'never calls removeItem');
+    run.cancelTimers();
```

Comment:
why cancel timers after the assertion?

Reply:
The test state won't settle without it because there's an infinite ember-concurrency task - could do it before the assert too, I'll move it up because I think the test will read better that way.